orchestrator/cosmos_peggy/src/query.rs

```diff
@@ -0,0 +1,126 @@
+use crate::types::*;
+use clarity::Uint256;
+use contact::client::Contact;
+use contact::jsonrpc::error::JsonRpcError;
+use contact::types::ResponseWrapper;
+use contact::types::TypeWrapper;
+use deep_space::address::Address;
+
+/// Get the latest valset recorded by the peggy module. If no valset has ever been created
+/// you will instead get a blank valset at height 0. Any value above this may or may not
+/// be a complete valset and it's up to the caller to interpret the response.
+pub async fn get_peggy_valset(contact: &Contact) -> Result<ResponseWrapper<Valset>, JsonRpcError> {
+    let none: Option<bool> = None;
+    let ret: Result<ResponseWrapper<ValsetUnparsed>, JsonRpcError> = contact
+        .jsonrpc_client
+        .request_method("peggy/current_valset", none, contact.timeout, None)
+        .await;
+    match ret {
+        Ok(val) => Ok(ResponseWrapper {
+            height: val.height,
+            result: val.result.convert(),
+        }),
+        Err(e) => Err(e),
+    }
+}
+
+/// get the valset for a given nonce (block) height
+pub async fn get_peggy_valset_request(
+    contact: &Contact,
+    nonce: Uint256,
+) -> Result<ResponseWrapper<Valset>, JsonRpcError> {
+    let none: Option<bool> = None;
+    let ret: Result<ResponseWrapper<TypeWrapper<ValsetUnparsed>>, JsonRpcError> = contact
+        .jsonrpc_client
+        .request_method(
+            &format!("peggy/valset_request/{}", nonce),
+            none,
+            contact.timeout,
+            None,
+        )
+        .await;
+    match ret {
+        Ok(val) => Ok(ResponseWrapper {
+            height: val.height,
+            result: val.result.value.convert(),
+        }),
+        Err(e) => Err(e),
+    }
+}
+
+/// This hits the /pending_valset_requests endpoint and will provide the oldest
+/// validator set we have not yet signed.
```

Comment:
Your assumption isn't wrong from a relayer standpoint I think I'll adapt my code to work the way you describe rather than change the functionality of the endpoint.

Slashing is in theory something we would want to avoid. By going back and signing valsets or batches we 'missed' within the allowed window. But since we're a long way from actually implementing slashing anyway I think we can drop that need for now.